src/submission/submission-updates-messenger.ts

```diff
@@ -0,0 +1,81 @@
+import { Kafka, Producer } from 'kafkajs';
+import { loggerFor } from '../logger';
+const L = loggerFor(__filename);
+
+let instance: SubmissionUpdatesMessenger;
+
+export interface SubmissionUpdatesMessenger {
+  sendProgramUpdatedMessage(programId: string): Promise<void>;
+  closeOpenConnections(): Promise<void>;
+}
+
+class DummyMessenger implements SubmissionUpdatesMessenger {
+  sendProgramUpdatedMessage = async (programId: string) => {
+    L.info('DummyMessenger called to send message to broker for updated program: ' + programId);
+    return;
+  };
+  closeOpenConnections = async () => {
+    return;
+  };
+}
+
+class KafkaMessenger implements SubmissionUpdatesMessenger {
+  private producer: Producer;
+  private knownTopics: ExpectedTopicsMap;
+
+  constructor(clientId: string, brokers: string[], topics: ExpectedTopicsMap) {
+    this.producer = new Kafka({ clientId, brokers }).producer();
+    this.knownTopics = topics;
+  }
+
+  sendProgramUpdatedMessage = async (programId: string) => {
+    L.info('KafkaMessenger called to send message to broker for updated program: ' + programId);
+    await this.producer.connect().catch(this.producerErrorHandler);
+    await this.producer
+      .send({
+        topic: this.knownTopics.programUpdate,
+        messages: [
+          {
+            key: programId,
+            value: JSON.stringify({ programId }),
+          },
+        ],
+      })
+      .catch(this.producerErrorHandler);
+    await this.producer.disconnect().catch(this.producerErrorHandler);
+  };
+
+  closeOpenConnections = async () => {
+    await this.producer.disconnect().catch(this.producerErrorHandler);
+  };
+
+  private producerErrorHandler = (e: any) =>
+    L.error('Found producer error in Kafka messenger: ', e);
+}
+
+type ExpectedTopicsMap = {
+  programUpdate: string;
+};
+
+export type KafkaConfig = {
+  clientId: string;
+  brokers: string[];
+  expectedTopics: ExpectedTopicsMap;
+};
+
+export const getInstace = (): SubmissionUpdatesMessenger => {
```

Comment:
is there a specific reason to go with a singleton here ?

Reply:
I used singleton for schema manager and I kind of feel it was confusing

Reply:
as long as it will be easy to test  for you (point to a test container or something) this should be fine was just wondering

Reply:
I think I did it this way because I wanted to make sure it was only used for submission-update related messages. Also it made it easy to initialize at bootstrap